modules/builds-docker-credentials-private-registries.adoc

```diff
@@ -115,3 +116,18 @@ This example uses `pullSecret` in a Source build, but it is also applicable
 in Docker and Custom builds.
 ====
 endif::[]
+
+. You can also link the pull secret to the service account used by the build instead of specifying
+the `pullSecret` field. By default builds use the *builder* service account. The pull secret is
+automatically added to the build if the secret contains a credential that matches the repository
+hosting the build's input image.
++
+----
+$ oc secrets link builder dockerhub
+----
+
+[NOTE]
+====
+In some circumstances Docker strategy builds which use `Binary` source may not have pull secrets
```

Comment:
It does go through that path I noted ... I had to debug / diagnose this as part of triaging a recent BZ from QE, https://bugzilla.redhat.com/show_bug.cgi?id=1810920#c4 , that is suppose to be addressed by this PR

Reply:
ok, that's not specific to binary build.  You could have the same situation if you had a git repo that contained your Dockerfile, and did not specify the "From" image in your buildconfig.

Reply:
(you also have the same situation if you inline a Dockerfile in the buildconfig, i believe, because i don't think we parse your dockerfile and try to extract the From image to choose a secret, as far as i recall)

Reply:
basically pull secret matching only works if you explicitly tell us, in your buildconfig, via the From field, what image your build needs.  If it's implicit (defined in the Dockerfile) it doesn't work.

Reply:
sound goods ... circling back to the line we are commenting on here, I don't think this note as is sufficiently covers minimally the situation wrt binary build ... and if there is a broader scope, either the updated note should address that, or the note should be nuked in lieu a more generic statement along those lines somewhere here

Reply:
agree

Reply:
I'll update the note that for any `Docker` strategy build the buildConfig must have the `from:` field populated. This is a given for all `Source` strategy builds.

Reply:
Going back to https://bugzilla.redhat.com/show_bug.cgi?id=1810920 I see that QE created their BC by running `oc  new-build --binary --name prune --to prune:latest`

Seems a bit iffy that they'll realize they need to cross reference this note with the resulting BC from our CLI:

```
apiVersion: build.openshift.io/v1
kind: BuildConfig
metadata:
  annotations:
    openshift.io/generated-by: OpenShiftNewBuild
  creationTimestamp: "2020-03-12T07:24:38Z"
  labels:
    build: prune
  name: prune
  namespace: wzheng1
  resourceVersion: "129030"
  selfLink: /apis/build.openshift.io/v1/namespaces/wzheng1/buildconfigs/prune
  uid: 9be8bd7c-58d7-4b29-8aef-9cc8ebfbfa6a
spec:
  failedBuildsHistoryLimit: 5
  nodeSelector: null
  output:
    to:
      kind: ImageStreamTag
      name: prune:latest
  postCommit: {}
  resources: {}
  runPolicy: Serial
  source:
    binary: {}
    type: Binary
  strategy:
    dockerStrategy: {}
    type: Docker
  successfulBuildsHistoryLimit: 5
  triggers:
  - github:
      secret: pDP4dNb7u7DBAn6KCOmy
    type: GitHub
  - generic:
      secret: YVMKtFuDF90Frys48P4u
    type: Generic
status:
  lastVersion: 1
```

At first blush, seems like 
- perhaps we note at least short term that resulting binary BCs from `oc new-build` (and I'm guessing `oc new-app` as well) may need manual adjustment per this new note
- it may not be quick fix, but do we entertain augmenting the CLI to fill in that field (based on fetching and parsing of the Dockerfile it finds on the local system

Reply:
> it may not be quick fix, but do we entertain augmenting the CLI to fill in that field (based on fetching and parsing of the Dockerfile it finds on the local system

I think just doc'ing the limitation + implications is sufficient.